src/components/signup/actions.js

```diff
@@ -16,22 +16,33 @@ export const signupRequest = payload => ({
  * Triggers request to update redux store on success
  *
  * @function
- * @param {Object} payload An object containing signup status (success or failure) provided 
- * the endpoint is reachable
+ * @param { boolean} isSuccessful the success(true) state of signup
  * @return {Object} The {@link actionTypes.SIGNUP_SUCCESS SIGNUP_SUCCESS} action.
  */
-export const signupSuccess = payload => ({
-    payload,
+export const signupSuccess = isSuccessful => ({
+    isSuccessful,
     type: SIGNUP_SUCCESS,
+}); 
```

Comment:
are you ever passing `false` into `signupSuccess`. Or are you `signupFail` to handle when signup is a failure. Its either or not both. same with `signinSuccess`